go.mod

```diff
@@ -1,9 +1,175 @@
 module github.com/caddyserver/cache-handler
 
-go 1.16
+go 1.18
 
 require (
-	github.com/caddyserver/caddy/v2 v2.4.6
-	github.com/darkweak/souin v1.6.6
+	github.com/buraksezer/olric v0.4.7
```

Comment:
I think it would be nice to eventually have olric/etcd etc support split out into separate modules. For most users, this will just be bloat in the binary, and they could plug in these storage providers only if they need it. But that's for later, I guess.

Reply:
Essentially it means you need to set up an interface that each of these will implement, and use some default zero-dependency implementation if none is configured.

Basically, the cache module itself would be a host module, and the storage providers would be guest modules. See https://caddyserver.com/docs/extending-caddy for a refresher.